core/string/ustring.h

```diff
@@ -46,6 +46,8 @@ class CharStringT;
 /*  Utility Functions                                                    */
 /*************************************************************************/
 
+namespace std {
```

Comment:
This is peculiar but i guess it is the easiest way to handle this problem?

Reply:
Not just peculiar but [undefined behavior](https://en.cppreference.com/w/cpp/language/extending_std), apart from a few well-defined exceptions :thinking:

In practice it probably works, but maybe we can go for an alternative here?

Reply:
We're already straying from convention with this original implementation, so a rework might be better suited from a ground-up standpoint. Though that puts a good chunk of this PR as a whole into question; I'll give this some thought

Reply:
The easiest way would be to move `strlen` to `STring::strlen`. Although I wouldn't know why we can't just hook into `std`?